### PR TITLE
fix: removed default value from whitelistedNonPeerDependencies in json schema

### PR DESCRIPTION
**Pull request by devsull**
## I'm submitting a...

```
[X] Bug Fix
[ ] Feature
[ ] Other (Refactoring, Added tests, Documentation, ...)
```

## Checklist

- [X] Commit Messages follow the [Conventional Commits](https://conventionalcommits.org/) pattern
  - A feature commit message is prefixed "feat:"
  - A bugfix commit message is prefixed "fix:"
- [ ] Tests for the changes have been added

## Description

Ref: https://github.com/ng-packagr/ng-packagr/issues/1892

Removed the default value from deprecated property `whitelistedNonPeerDependencies` in the json schema.

## Does this PR introduce a breaking change?

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Comment by devsull** [comment]
It would be nice to write tests for this, but I'm not sure how to structure them, or test for these cases.... 😕

File: src/lib/ng-package/package.ts [hunk 1]
```diff
@@ -78,12 +78,13 @@ export class NgPackage {
     const alwaysInclude = ['tslib'];
 
     // remove in the future, after deprecation phase
-    const deprecatedPropList = this.whitelistedNonPeerDependencies;
+    let deprecatedPropList = this.whitelistedNonPeerDependencies;
```
Comment (skrtheboss) [inline src/lib/ng-package/package.ts:81]: Can be improoved: 

`const deprecatedPropList = this.whitelistedNonPeerDependencies ?? [];`

File: src/lib/ng-package/package.ts [hunk 2]
```diff
@@ -78,7 +78,7 @@ export class NgPackage {
     const alwaysInclude = ['tslib'];
 
     // remove in the future, after deprecation phase
-    const deprecatedPropList = this.whitelistedNonPeerDependencies;
+    const deprecatedPropList = this.whitelistedNonPeerDependencies ?? [];
```
Comment (alan-agius4) [inline src/lib/ng-package/package.ts:81]: Use the billable operator instead.

File: src/lib/ng-package/package.ts [hunk 3]
```diff
@@ -78,7 +78,7 @@ export class NgPackage {
...
 
     if(deprecatedPropList.length) {
```
Comment (alan-agius4) [inline src/lib/ng-package/package.ts:83]: `deprecatedPropList?.length`

**Comment by alan-agius4** [comment]
Actually thinking about this, it’s a bug in the schema validation which is found https://github.com/ng-packagr/ng-packagr/blob/85eba9dab48d84fb591514807150cacb61068b16/src/lib/ng-package/schema.ts#L50

**Comment by devsull** [comment]
@alan-agius4 I don't understand what the bug would be in what you mentioned above, about `schema.ts` ?

**Comment by devsull** [comment]
> Actually thinking about this, it’s a bug in the schema validation which is found
> 
> https://github.com/ng-packagr/ng-packagr/blob/85eba9dab48d84fb591514807150cacb61068b16/src/lib/ng-package/schema.ts#L50
How is the bug here?

**Comment by alan-agius4** [comment]
We should not display the warning when the value is defined by the user. 

That logic, is responsible for displaying the warning.

**Comment by alan-agius4** [comment]
Took a look at this and there is no easy way to fix this in the schema registry.

**Pull request merged**